src/backend/utils/activity/pgstat_function.c

```diff
@@ -79,7 +80,7 @@ pgstat_init_function_usage(FunctionCallInfo fcinfo,
 	PgStat_BackendFunctionEntry *pending;
 	bool		created_entry;
 
-	if (pre_function_call_hook)
+	if (pre_function_call_hook && IsTransactionState())
```

Comment:
Do we have a proper error message telling the customer that the client_encoding needs to be set inside a transaction?

Reply:
No , we don't, the IsTransactionState() for the pstack crash happened is already out of customer's top level transaction, in this case we don't need to return any error msg, just bypass this call_hook is enough